modules/sdk/src/main/scala/org/tessellation/sdk/domain/cluster/programs/Joining.scala

```diff
@@ -242,6 +243,13 @@ sealed abstract class Joining[F[_]: Async: GenUUID: SecurityProvider: KryoSerial
 
   private def validateHandshake(registrationRequest: RegistrationRequest, remoteAddress: Option[Host]): F[Unit] =
     for {
+
+      versionHash <- BuildInfo.version.hashF
```

Comment:
I think that the version should be passed from `version` property of `TessellationIOApp`. Otherwise we're just comparing the version of the `shared` library.

Reply:
`version` property is exactly the same one (taken from BuildInfo from shared module)

Reply:
Yes, but that's just a coincidence. If someone would build a different app based on `TessellationIOApp`, then it would compare the version of the `shared` library rather than an app.